docs/components/fab.md

```diff
@@ -0,0 +1,369 @@
+<!--docs:
+title: "Floating action button"
+layout: detail
+section: components
+excerpt: "A floating action button (FAB) represents the primary action of a screen."
+iconId: 
+path: /catalog/floating-action-button/
+-->
+
+# Floating action buttons
+
+A floating action button (FAB) represents the primary action of a screen.
+
+There are three types of FABS:
+
+1. [Regular FABs](#regular-fabs)
+2. [Mini FABs](#mini-fabs)
+3. [Extended FABs](#extended-fabs)
+
+![Regular FAB example](assets/FAB_types.png)
+
+## Using FABs
+
+A FAB performs the primary, or most common, action on a screen. It appears in front of all screen content, typically as a circular shape with an icon in its center. 
+
+Before you can use Material FABs, you need to import the Material Components package for Flutter: 
+```dart
+package:flutter/material.dart.
+```
+
+You need to be using a [`MaterialApp`](https://api.flutter.dev/flutter/material/MaterialApp-class.html).
+
+For more information on getting started with the Material for Flutter, go to the Flutter [Material library](https://api.flutter.dev/flutter/material/material-library.html) page.
+
+
+### Making FABs accessible
+
+Flutter's APIs support accessibility setting for large fonts, screen readers, and sufficient contrast. For more information, go to Flutter's [accessibility](https://flutter.dev/docs/development/accessibility-and-localization/accessibility) and [internationalization](https://flutter.dev/docs/development/accessibility-and-localization/internationalization) pages.
+
+For more guidance on writing labels, go to [our page on how to write a good accessibility label](https://material.io/design/usability/accessibility.html#writing).
+
+
+
+## Regular FABs
+
+
+Regular FABs are FABs that are not expanded and are a regular size.
+
+`FloatingActionButton`
+* [Class description](https://api.flutter.dev/flutter/material/FloatingActionButton-class.html)
+* [GitHub source](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/floating_action_button.dart)
+* [Dartpad Demo](https://dartpad.dev/embed-flutter.html?gh_owner=material-components&gh_repo=material-components-flutter&gh_path=docs/components/dartpad/fabs/regular&gh_ref=develop)
+
+
+
+The following example shows a regular FAB.
+
+!["Regular FAB example for Flutter with white '+' icon over a black background."](assets/fabs/regular_fab.png)
+
+```dart
+FloatingActionButton(
+  backgroundColor: Colors.black,
+  foregroundColor: Colors.white,
+  onPressed: () {
+    // Respond to button press
+  },
+  child: Icon(Icons.add),
+)
+```
+
+### Anatomy and key properties
+
+![Regular FAB anatomy diagram](assets/FAB_anatomy-long.png)
+
+A regular FAB has a container and an icon.
+
+1. Container
+1. Icon
+
+#### Container attributes
+
+&nbsp;         | Property
+-------------- | -----------------------
+**Color** | `backgroundColor`
+**Stroke color** | N/A
```

Comment:
I believe this can be accomplished with `FloatingActionButton.shape`.

Reply:
true, good catch